src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -567,4 +571,64 @@ public virtual void PrintMessageNormalized(TextWriter writer, string message, bo
         else if (!removeLastNewLine)
             writer.WriteLine();
     }
+
+    /// <summary>
+    /// Adds a new runtime option. Throws if option already exists.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    public void AddOption<T>(string name, T value)
+    {
+        Options.Add(name, value);
+    }
+
+    /// <summary>
+    /// Trys to add a new runtime option.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    /// <returns>True if successful. False otherwise.</returns>
+    public bool TryAddOption<T>(string name, T value)
+    {
+        if (Options.ContainsKey(name))
+            return false;
+        AddOption<T>(name, value);
+        return true;
+    }
+
+    public void AddOrOverwriteOption<T>(string name, T value)
+    {
+        if (!Options.ContainsKey(name))
+            AddOption<T>(name, value);
+        else
+            Options[name] = value;
+    }
+
+    public T GetOption<T>(string name)
+    {
+        return (T)Options[name];
+    }
+
+    public T GetOptionOrDefault<T>(string name)
+    {
+        if (!Options.ContainsKey(name))
+            AddOption<T>(name, default);
+
+        return GetOption<T>(name);
+    }
+
+    public void RemoveOption(string name)
+    {
+        Options.Remove(name);
+    }
+
+    public bool TryRemoveOption(string name)
+    {
+        if (!Options.ContainsKey(name))
+            return false;
+        Options.Remove(name);
+        return true;
```

Comment:
Dictiionary.Remove already return bool. you can just implement this as `=> Options.Remove(name)`

Reply:
Also, do you think we need `RemoveOptions` method at all?

Reply:
Think I need separate remove and try remove then? since they would basically be the same? Or should I just have a remove that returns a bool? Thats what I'll go with for now.

Reply:
Having `Remove` return `bool` will be consistent with the `Dictionary` which can be ok, but I prefer to name it `TryRemove`. So, we end up with one method for removing.

Reply:
What if we only have a Remove and not a `TryRemove`? That way we only have 1 method but also are consistent with the Dictionary?

Reply:
I am fine either way.